### ISSUE TITLE
Probably a stupid question... how do I enter a carriage return when the autocomplete pop up is present?

### ISSUE THREAD

**Issue by downhiller** [nvim-lua/kickstart.nvim#168]
This is such a common issue that there must be a solution, that my googling isn't throwing up.

Suppose I've typed this (PHP):

    function myFund(string $foo): array

And I hit "enter" after array to put the `{` on the next line

    function myFund(string $foo): array
    {

Instead of allowing me to do that, it auto-completes `array` to `array_remove` (or whatever is at the top of the auto-complete list)

So far the best option seems to be `esc` `o`.  Quits the autocomplete, goes back to normal mode, then puts me on the next line.  It's not exactly intuitive or easy keys to press.

I think I'd prefer if `esc` just closed the auto-complete panel, rather than went back to normal mode.  Or if (say) `shift` + `enter` would work for a CR.

Please feel free to tell me what I'm doing wrong, and/or what your solution is!

Thank you

**Comment by dhall3y** [comment]
Hi, to get `esc` to close the auto complete panel you can add this line in the cmp.setup near the end of the init.lua :
```   ['<Esc>'] = cmp.mapping.close(),```

you should get something that looks like this: 
``` 
  mapping = cmp.mapping.preset.insert {
    ['<C-d>'] = cmp.mapping.scroll_docs(-4),
    ['<Esc>'] = cmp.mapping.close(),
    ['<C-f>'] = cmp.mapping.scroll_docs(4),
    ['<C-Space>'] = cmp.mapping.complete(),
    ['<CR>'] = cmp.mapping.confirm {
      behavior = cmp.ConfirmBehavior.Replace,
      select = true,
    },
```
By default `<C-e>` should close the panel, so you can add this to bind `shift + enter` to do what you want.
``` vim.keymap.set('i', '<S-CR>', '<C-e><CR>') ```

**Comment by downhiller** [comment]
Thanks @dhall3y .  I've actually already added the `esc` keymapping to close the autocomplete.  I tried shift+enter as well but esc feels more natural.

I was curious though.  kickstart.nvim is supposed to be a good baseline, so I wondered if there was something I was missing.  Some other way to close the autocomplete panel which I hadn't noticed.

Should I assume from your response that there's not, and you either have to suck up esc/o or add your own keybindings to make it work how one wants?

I'm a sole dev and while I know the whole point of nvim is customisability, I prefer to make sure I'm not missing something obvious, or a better way of doing things!
